### ISSUE TITLE
New query for users GetAvailableUsersToAdd

### ISSUE THREAD

**Issue by ChristosTanailidis** [Lemas97/Sousou-Api#16]
New query thats fetches all users that are available for the logged user to add as friends.

Differences with the normal `GetUers` :

- `GetAvailableUsersToAdd` will not include the logged users as an option
- `GetAvailableUsersToAdd` will not include friends of the user
- `GetAvailableUsersToAdd` will include a **nullable** field `pending?: boolean`. This field will be true when a request is already made on that user, to prevent spamming.

**Comment by ChristosTanailidis** [comment]
What is needed: 
```graphql
query GetAvailableUsersToAdd($paginatedData: PaginatedInputData!) {
    getAvailableUsersToAdd(paginatedData: $paginatedData) {
      data {
         pending
         ...
      }
      total
    }
}
```
Requirements:
1. Query parameters will be same as in `GetUsers` (pagination is needed).
2. An extra field can be added in the query with `boolean` or `string ('pending' | 'available' | 'all')` type value. This is optional anyways and not needed right now.
3. The pending field is needed in the data object on the query response.

Example of usage in front:

- admin@sousous.gr user is available (pending = false) to add.
- **user#6057** is unavailable to add because a friend request is already sent to him (pending = true).
 
![image](https://user-images.githubusercontent.com/32941481/186009113-699db165-2239-41a2-a750-56e9ade8fbb4.png)

**Comment by Lemas97** [comment]
closes at f6738dcc84a4c03fa277cd183062b27f9f457155

**Comment by ChristosTanailidis** [comment]
GetAvailableUsersToAdd query after friend request is been answered (_accepted_)

- Pending remains as `true`
- User is still getting fetched (_a friend should not be fetched on this query_)

![image](https://user-images.githubusercontent.com/32941481/188712560-bff907d0-ed08-4783-b1f1-390f285aa90a.png)

**Comment by Lemas97** [comment]
First bullet closed at 9f9de8eedd0893f9b08120b02ac8521a613a90ce
The second bullet seems to work fine to me :/
